common/datavalues/src/vec.rs

```diff
@@ -265,14 +277,49 @@ impl<T> AlignedVec<T> {
         self.inner.extend(iter);
         let added = self.len() - len_before;
 
-        assert_eq!(added, cap)
+        if added != cap {
+            eprintln!("size hint was incorrect, this is UB. aborting");
+            std::process::abort()
+        }
     }
 }
 
-impl<T> Default for AlignedVec<T> {
+impl<T: ArrowNativeType> Default for AlignedVec<T> {
     fn default() -> Self {
         // Be careful here. Don't initialize with a normal Vec as this will cause the wrong deallocator
         // to run and SIGSEGV
         Self::with_capacity_aligned(0)
     }
 }
+
+#[cfg(test)]
```

Comment:
It;s better to make a new file vec_test.rs here

Reply:
Ok